frontend/src/components/auth/Logout.tsx

```diff
@@ -1,22 +1,53 @@
-import React, { useContext } from "react";
-import authAPIClient from "../../APIClients/AuthAPIClient";
+import React, { useContext, useEffect } from "react";
+import { gql, useMutation } from "@apollo/client";
 import AuthContext from "../../contexts/AuthContext";
+import AUTHENTICATED_USER_KEY from "../../constants/AuthConstants";
+
+interface LogoutResponse {
+  ok: Boolean;
+}
+
+const LOGOUT = gql`
+  mutation Logout($userId: ID!) {
+    logout(userId: $userId) {
+      ok
+    }
+  }
+`;
 
 const Logout = () => {
   const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const [logout, { data, error }] = useMutation<{ logout: LogoutResponse }>(
+    LOGOUT,
+  );
 
-  const onLogOutClick = async () => {
```

Comment:
Let's keep with the `onLogOutClick` pattern rather than declaring the `onClick` fn inline and using `useEffect`. `Login.tsx` and `ResetPassword.tsx` are good examples.

`error` may actually come in handy in `onLogOutClick`, so feel free to keep that in